test/snapshot_test.go

```diff
@@ -77,45 +80,53 @@ func (suite *SnapshotWorkflowSuite) TestCreateUpdateSnapshot() {
 	volumeID := suite.createVolume(volumeName, "")
 	offlineSnapshotName := "snapoffline"
 	newOfflineSnapshot := &nimbleos.Snapshot{
-		Name:  param.NewString(offlineSnapshotName),
-		VolId: param.NewString(volumeID),
+		Name:  &offlineSnapshotName,
+		VolId: &volumeID,
 	}
 	_, err := suite.snapshotService.CreateSnapshot(newOfflineSnapshot)
-	assert.Nilf(suite.T(), err, "Failed to create snapshot")
+	assert.Nilf(suite.T(), err, "Failed to create snapshot: %v", offlineSnapshotName)
 
 	getSnapResp, err := suite.snapshotService.GetSnapshotByName(volumeName)
-	assert.Nilf(suite.T(), err, "Failed to get snapshot")
+	assert.Nilf(suite.T(), err, "Failed to get snapshot by name: %v", offlineSnapshotName)
```

Comment:
This message is misleading.  It actually tried to retrieve the snapshot by the _volume_ name, so it would be better to include volumeName in the message.  (The GetSnapshotByName function name is misleading too.)